Dockerfile.osx

```diff
@@ -8,39 +8,26 @@ RUN if [ -z "${mono_version}" ]; then echo -e "\n\nargument mono-version is mand
       automake autoconf bzip2-devel clang libicu-devel libtool libxml2-devel llvm-devel openssl-devel yasm && \
     git clone --progress https://github.com/tpoechtrager/osxcross.git && \
     cd /root/osxcross && \
-    git checkout 61925a29c404fc547ce9d51eb2ee4a09bd6533da && \
+    git checkout 447cf3b3ea4323d24648f5f7f775f5977a0d15bd && \
     ln -s /root/files/MacOSX11.1.sdk.tar.xz /root/osxcross/tarballs && \
-    UNATTENDED=1 ./build.sh
+    patch -p1 < /root/files/patches/osxcross-pr284-compiler-rt.patch && \
+    UNATTENDED=1 ./build.sh && \
+    ENABLE_COMPILER_RT_INSTALL=1 ./build_compiler_rt.sh
 
 ENV OSXCROSS_ROOT=/root/osxcross
 ENV PATH="/root/osxcross/target/bin:${PATH}"
 
 RUN cp -a /root/files/${mono_version} /root && \
     cd /root/${mono_version} && \
-    export CMAKE=/root/osxcross/target/bin/x86_64-apple-darwin20.2-cmake && \
-    NOCONFIGURE=1 ./autogen.sh && \
-    ./configure --prefix=/root/dependencies/mono \
-        --build=x86_64-linux-gnu \
-        --host=x86_64-apple-darwin20.2 \
-        --disable-boehm \
-        --disable-mcs-build \
-        --with-tls=pthread \
-        --disable-dtrace \
-        --disable-executables \
-        mono_cv_uscore=yes \
-        _lt_dar_can_shared=yes \
-        CC=o64-clang \
-        CXX=o64-clang++ && \
-    make -j && \
-    make install && \
+    patch -p1 < /root/files/patches/mono-btls-cmake-wrapper.patch && \
+    export MONO_SOURCE_ROOT=/root/${mono_version} && \
+    export OSXCROSS_SDK=20.2 && \
+    cd /root/${mono_version}/godot-mono-builds && \
+    python3 osx.py configure -j --target=x86_64 --target=arm64 && \
+    python3 osx.py make -j --target=x86_64 --target=arm64 && \
+    python3 bcl.py make --product=desktop && \
+    python3 osx.py copy-bcl --target=x86_64 --target=arm64 && \
```

Comment:
@neikeq While working on this, I was a bit confused that desktop platforms have `copy-bcl` but Android/iOS/JavaScript don't seem to need it. It's not a big deal but that made me wonder if I'm missing something in the mobile/web build, or if this `copy-bcl` isn't redundant for desktop.

Also, how do the `--target` arguments impact `copy-bcl`? Isn't the bcl cross-platform and arch-independent (aside from `desktop-win32`)?

Reply:
What `copy-bcl` does is to copy the previously built BCL into the runtime install folder in the subdir `mono_root/lib/mono/4.5/`. That's where SCons will search for it when building the Godot editor, which is desktop only hence why there's no such option for Android, iOS, etc.
I don't know if you were doing this step manually before with the `cp` command, but an alternative if you prefer it is to skip the step entirely and just pass the BCL path to SCons via `mono_bcl=<path>`.

Reply:
Ah I see. The version of the BCL I use is probably the one built in `Dockerfile.mono`, which is the only image where I still have a manual build process:
https://github.com/godotengine/build-containers/blob/containers-overhaul-linux-sdk-macos-arm64/Dockerfile.mono#L11-L14

So I guess I don't need the `copy-bcl` in any of the desktop containers?

Might also be worth updating `Dockerfile.mono` to also use `godot-mono-builds` and do the install in `/usr` (which is then used by the manually installed msbuild RPMs), if that's possible?

Reply:
> So I guess I don't need the `copy-bcl` in any of the desktop containers?

I'm looking at the changes in this PR again and I think you actually need it. If I understand the old code in this file correctly, it used to build both the Mono runtime and the BCL and install them at `/root/dependencies/mono` and then created a softlink in `/usr` pointing at that. If that's the case, then for the new code to do the equivalent you need `copy-bcl` as well.

But I'm not really familiar with the build containers and it's a bit confusing so can't tell for sure.

Reply:
Ah I see, so the `bcl.py` build gives me this:
```
/root/mono-installs/desktop-bcl/net_4_x/
```

But I still need the `copy-bcl` step for desktop platforms (which are compilation hosts) to copy it into e.g. `/root/mono-installs/desktop-osx-x86_64-release/lib/mono/`.
And thus the `--target` makes sense so that it's copied for each desktop target.

Reply:
> And thus the `--target` makes sense so that it's copied for each desktop target.

Right, the target is needed.

BTW, IIRC this option was added by a contributor that needed it. I couldn't imagine it would end up being needed by the official build scripts.